mobile_verifier/src/boosting_oracles/data_sets.rs

```diff
@@ -0,0 +1,482 @@
+        pool: &PgPool,
+        data_set_type: DataSetType,
+        since: Option<DateTime<Utc>>,
+    ) -> sqlx::Result<Option<NewDataSet>> {
+        sqlx::query_as(
+                "SELECT filename, time_to_use, status FROM data_sets WHERE status != 'processed' AND data_set = $1 AND COALESCE(time_to_use > $2, TRUE) AND time_to_use <= $3 ORDER BY time_to_use DESC LIMIT 1"
+            )
+            .bind(data_set_type)
+            .bind(since)
+            .bind(Utc::now())
+            .fetch_optional(pool)
+            .await
+    }
+
+    pub async fn set_data_set_status(
+        pool: &PgPool,
+        filename: &str,
+        status: DataSetStatus,
+    ) -> sqlx::Result<()> {
+        sqlx::query("UPDATE data_sets SET status = $1 WHERE filename = $2")
+            .bind(status)
+            .bind(filename)
+            .execute(pool)
+            .await?;
+        Ok(())
+    }
+
+    pub async fn fetch_time_of_latest_processed_data_set(
+        pool: &PgPool,
+        data_set_type: DataSetType,
+    ) -> sqlx::Result<Option<DateTime<Utc>>> {
+        sqlx::query_scalar(
+                "SELECT time_to_use FROM data_sets WHERE status = 'processed' AND data_set = $1 ORDER BY time_to_use DESC LIMIT 1"
+            )
+            .bind(data_set_type)
+            .fetch_optional(pool)
+            .await
+    }
+
+    /// Check if there are any pending or downloaded files prior to the given reward period
+    pub async fn check_for_unprocessed_data_sets(
+        pool: &PgPool,
+        period_end: DateTime<Utc>,
+    ) -> sqlx::Result<bool> {
+        sqlx::query_scalar(
+            "SELECT COUNT(*) > 0 FROM data_sets WHERE time_to_use <= $1 AND status != 'processed'",
+        )
+        .bind(period_end)
+        .fetch_one(pool)
+        .await
+    }
+}
+
+#[derive(FromRow)]
+pub struct UnassignedHex {
+    uuid: uuid::Uuid,
+    #[sqlx(try_from = "i64")]
+    hex: u64,
+    signal_level: SignalLevel,
+    signal_power: i32,
+}
+
+impl UnassignedHex {
+    pub fn fetch_all(pool: &PgPool) -> impl Stream<Item = sqlx::Result<Self>> + '_ {
+        sqlx::query_as("SELECT uuid, hex, signal_level, signal_power FROM hexes").fetch(pool)
+    }
+
+    pub fn fetch_unassigned(pool: &PgPool) -> impl Stream<Item = sqlx::Result<Self>> + '_ {
+        sqlx::query_as(
+            "SELECT
+                uuid, hex, signal_level, signal_power
+            FROM
+                hexes
+            WHERE
+                urbanized IS NULL
+                OR footfall IS NULL
+                OR landtype IS NULL",
+        )
+        .fetch(pool)
+    }
+}
+
+pub async fn set_oracle_boosting_assignments<'a>(
+    unassigned_hexes: impl Stream<Item = sqlx::Result<UnassignedHex>>,
+    data_sets: &HexBoostData<impl HexAssignment, impl HexAssignment, impl HexAssignment>,
+    pool: &'a PgPool,
+) -> anyhow::Result<impl Iterator<Item = proto::OracleBoostingReportV1>> {
+    const NUMBER_OF_FIELDS_IN_QUERY: u16 = 5;
```

Comment:
This value looks to be from an old version of this function. Can we make sure no updates were missed from what's on master?

Reply:
Nice catch